File: source/code/nodejs/authentication/refresh_access_token_req.js
var refreshGrantRequest = OauthRequests.NewRefreshGrantRequest()
    .SetRefreshToken(refresh_token)
    .Build();

var grantResponse = await app.NewRefreshGrantAuthenticator()
    .AuthenticateAsync(refreshGrantRequest);


var authRequest = {
  refresh_token: refresh_token
};

var authenticator = new stormpath.OAuthRefreshTokenGrantRequestAuthenticator(application);

authenticator.authenticate(authRequest, function (err, authResult) {
  if (err) {
    return console.error(err);
  }

  console.log('Got access token!', authResult.accessToken.compact());
});

File: source/code/java/authentication/refresh_access_token_req.java
OAuthRefreshTokenRequestAuthentication request =
    OAuthRequests.OAUTH_REFRESH_TOKEN_REQUEST
        .builder()
        .setRefreshToken(refreshTokenString)
        .build();

OAuthGrantRequestAuthenticationResult authenticationResult =
    Authenticators.OAUTH_REFRESH_TOKEN_REQUEST_AUTHENTICATOR
        .forApplication(application)
        .authenticate(request);
